File: math.fraction/src/main/java/math/fraction/Fraction.java
package math.fraction;

import static java.lang.Integer.max;

public class Fraction {
    private final int numerator;
    private final int denominator;

    public Fraction(int integerValue) {
        this(integerValue,1);
//        this.numerator = integerValue;
//        this.denominator =1;
    }
    public Fraction(int numerator, int denominator) {
        this.numerator = numerator;
        this.denominator = denominator;
    }

    public Fraction plus(Fraction otherFraction) {
        int deno =0;
        Fraction fraction;

        if (  otherFraction.denominator !=0 &&
                ((otherFraction.denominator%this.denominator) ==0) ){
            deno = max(otherFraction.denominator, this.denominator);

            fraction = new Fraction((deno/this.denominator*this.numerator) +
                    otherFraction.numerator,deno);

        } else if(otherFraction.denominator !=0 && (otherFraction.denominator%this.denominator) !=0){
            deno = (otherFraction.denominator * this.denominator);
            fraction = new Fraction((deno/this.denominator*this.numerator) +
                    deno/otherFraction.denominator*otherFraction.numerator,deno);
        }
        else{
            fraction =  new Fraction(this.numerator +
                    otherFraction.numerator,denominator);
        }
        return fraction;
    }

    public int intValue() {
        return numerator;
    }

    public int getNumerator() {
        return numerator;
    }

    public int getDenominator() {
        return denominator;
    }

    // con este metodo equals, sobreescribo el equals que usan los test
    @Override
    public boolean equals(Object other){
        if (other instanceof Fraction) {
            Fraction that = (Fraction) other;
            return this.numerator == that.numerator
                    && this.denominator == that.denominator;
        }
        return false;
    }
    // con este metodo en el equals anterior le digo que la salida
    // del objeto Fraction sea String con este formato
    @Override
    public String toString(){
        return String.format("%d/%d",numerator,denominator);
    }
}

File: math.fraction/src/main/kotlin/Fraction.kt
import java.lang.Integer.max

class Fraction{
    val integerValue: Int
    private var denominator=0 // una val no puede ser reasignada,
                             // var si (es mutable)

    constructor(integerValue: Int){
        this.integerValue = integerValue
    }

    constructor(integerValue: Int, denominator: Int){
        this.integerValue = integerValue
        this.denominator = denominator
    }

    fun plus(otherfraction: Fraction):Fraction{

        if ((otherfraction.denominator%this.denominator)==0){
            val deno = max(otherfraction.denominator,this.denominator)
            return (Fraction((this.integerValue*(deno/this.denominator))+
                    otherfraction.integerValue,deno))
        }

        return Fraction(this.integerValue +
                otherfraction.integerValue, denominator)
    }

    fun toIntValue():Int{
        return this.integerValue
    }

    fun getNumerator(): Int {
        return this.integerValue
    }

    fun getDenominator(): Int {
        return denominator
    }
}